src/naming_screen.c

```diff
@@ -0,0 +1,2400 @@
+    .oam = &gOamData_858BFFC,
+    .anims = gSpriteAnimTable_858C0B0,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = SpriteCallbackDummy
+};
+
+static const struct SpriteTemplate gUnknown_83E25A4 = {
+    .tileTag = 0x0004,
+    .paletteTag = 0x0004,
+    .oam = &gOamData_858BFEC,
+    .anims = gSpriteAnimTable_858C0B0,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = SpriteCallbackDummy
+};
+
+static const struct SpriteTemplate gUnknown_83E25BC = {
+    .tileTag = 0x0000,
+    .paletteTag = 0x0006,
+    .oam = &gOamData_858BFEC,
+    .anims = gSpriteAnimTable_858C0B0,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = SpriteCallbackDummy
+};
+
+static const struct SpriteTemplate gUnknown_83E25D4 = {
+    .tileTag = 0x0001,
+    .paletteTag = 0x0007,
+    .oam = &gOamData_858BFEC,
+    .anims = gSpriteAnimTable_858C0B0,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = SpriteCallbackDummy
+};
+
+static const struct SpriteTemplate gUnknown_83E25EC = {
+    .tileTag = 0x0007,
+    .paletteTag = 0x0005,
+    .oam = &gOamData_858BFF4,
+    .anims = gSpriteAnimTable_858C0B4,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = sub_809E700
+};
+
+static const struct SpriteTemplate sSpriteTemplate_InputArrow = {
+    .tileTag = 0x000A,
+    .paletteTag = 0x0003,
+    .oam = &gOamData_858BFEC,
+    .anims = gSpriteAnimTable_858C0B0,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = sub_809E7F0
+};
+
+static const struct SpriteTemplate sSpriteTemplate_Underscore = {
+    .tileTag = 0x000B,
+    .paletteTag = 0x0003,
+    .oam = &gOamData_858BFEC,
+    .anims = gSpriteAnimTable_858C0B0,
+    .images = NULL,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = sub_809E83C
+};
+
+static const struct SpriteTemplate gUnknown_83E2634 = {
+    .tileTag = 0xFFFF,
+    .paletteTag = 0x0000,
+    .oam = &gOamData_858BFEC,
+    .anims = gSpriteAnimTable_858C0BC,
+    .images = gUnknown_0858C080,
+    .affineAnims = gDummySpriteAffineAnimTable,
+    .callback = SpriteCallbackDummy
+};
+
+static const u8 *const sNamingScreenKeyboardText[KBPAGE_COUNT][KBROW_COUNT] = {
+    [KBPAGE_LETTERS_LOWER] = {
+        gText_NamingScreenKeyboard_abcdef,
+        gText_NamingScreenKeyboard_ghijkl,
+        gText_NamingScreenKeyboard_mnopqrs,
+        gText_NamingScreenKeyboard_tuvwxyz
+    },
+    [KBPAGE_LETTERS_UPPER] = {
+        gText_NamingScreenKeyboard_ABCDEF,
+        gText_NamingScreenKeyboard_GHIJKL,
+        gText_NamingScreenKeyboard_MNOPQRS,
+        gText_NamingScreenKeyboard_TUVWXYZ
+    },
+    [KBPAGE_SYMBOLS] = {
+        gText_NamingScreenKeyboard_01234,
+        gText_NamingScreenKeyboard_56789,
+        gText_NamingScreenKeyboard_Symbols1,
+        gText_NamingScreenKeyboard_Symbols2
+    },
+};
+
+// FIXME: Sync with Emerald
```

Comment:
What do you mean?

Reply:
the gfx need to be reorganized to sync up with emerald